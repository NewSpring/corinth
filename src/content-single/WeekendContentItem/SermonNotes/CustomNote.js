import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { AnalyticsConsumer } from '@apollosproject/ui-analytics';
import {
  Icon,
  TextInput,
  BodyText,
  styled,
  Touchable,
  PaddedView,
} from '@apollosproject/ui-kit';

const StyledTextInput = styled(({ theme }) => ({
  borderWidth: 0.5,
  borderColor: theme.colors.shadows.default,
  paddingHorizontal: theme.sizing.baseUnit * 0.5,
  paddingTop: theme.sizing.baseUnit,
  paddingBottom: theme.sizing.baseUnit,
  textAlignVertical: 'top',
}))(TextInput);

const AddNoteView = styled({
  flex: 1,
  flexDirection: 'row',
})(PaddedView);

const CustomNote = ({ onChange }) => {
  const [hasBox, showBox] = useState(false);
  const [note, setNote] = useState('');
  return hasBox ? (
    <StyledTextInput
      multiline
      returnKeyType={'default'}
      value={note}
      scrollEnabled={false}
      onChangeText={(text) => {
        setNote(text);
        onChange(text);
      }}
    />
  ) : (
    <AnalyticsConsumer>
      {({ track }) => (
        <Touchable
          onPress={() => {
            showBox(true);
            track({ eventName: 'Added Custom Note' });
          }}
        >
          <AddNoteView horizontal={false}>
            <Icon name={'add'} size={24} />
            <PaddedView vertical={false}>
              <BodyText>Add a Note</BodyText>
            </PaddedView>
          </AddNoteView>
        </Touchable>
      )}
    </AnalyticsConsumer>
  );
};

CustomNote.propTypes = {
  onChange: PropTypes.func,
};

export default CustomNote;
