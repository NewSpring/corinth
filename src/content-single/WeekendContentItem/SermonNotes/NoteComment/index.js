import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import NoteComment from './NoteComment';
import SAVE_NOTES_COMMENT from './saveNotesComment';

const NoteCommentConnected = ({ contentID, noteID, onChange, initialText }) => {
  const [timer, setTimer] = useState(null);
  const [textToSave, setText] = useState(initialText);
  const handleChange = (saveComment) => (newText) => {
    setText(newText);
    onChange(newText);

    // autosave after 10 sec
    // check for existing timeout, clear it, and set a new one
    clearTimeout(timer);
    const newTimer = setTimeout(() => {
      console.count('saving');
      saveComment();
    }, 10000);
    setTimer(newTimer);
  };
  return (
    <Mutation
      mutation={SAVE_NOTES_COMMENT}
      variables={{ contentID, noteID, text: textToSave }}
    >
      {(saveComment) => (
        <NoteComment
          onChange={handleChange(saveComment)}
          initialText={initialText}
        />
      )}
    </Mutation>
  );
};

NoteCommentConnected.propTypes = {
  contentID: PropTypes.string,
  noteID: PropTypes.string,
  onChange: PropTypes.func,
  initialText: PropTypes.string,
};

export default NoteCommentConnected;
