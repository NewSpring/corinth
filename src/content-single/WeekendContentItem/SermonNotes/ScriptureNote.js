import React from 'react';
import { BodyText } from '@apollosproject/ui-kit';

const ScriptureNote = () => <BodyText>ScriptureNote Note</BodyText>;

export default ScriptureNote;
