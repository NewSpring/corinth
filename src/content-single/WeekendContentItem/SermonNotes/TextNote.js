import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity } from 'react-native';
import { H4, BodyText, PaddedView } from '@apollosproject/ui-kit';
import { AnalyticsConsumer } from '@apollosproject/ui-analytics';

const TextNote = ({ text, isHeader }) => {
  const [isPressed, press] = useState(false);
  const blanksRegex = /__(.*)__/gm;
  // TODO pull from API, called hiddenText
  const textWithBlanks = text.replace(blanksRegex, (match, p1) =>
    '_'.repeat(p1.length)
  );
  // TODO pull from API, called simpleText, use for ScriptureNotes too
  const textNoBlanks = text.replace(blanksRegex, (match, p1) => p1);

  // choose to show bold or not
  const TextComponent = isHeader ? H4 : BodyText;

  // choose to show blanks or not
  const Note = () => (
    <TextComponent>{isPressed ? textNoBlanks : textWithBlanks}</TextComponent>
  );
  return (
    <AnalyticsConsumer>
      {({ track }) => (
        <TouchableOpacity
          onPress={() => {
            press(true);
            track({ eventName: 'Clicked Fill-in-the-blank' });
          }}
          disabled={isPressed || !text.match(blanksRegex)}
        >
          <PaddedView horizontal={false}>
            <Note />
          </PaddedView>
        </TouchableOpacity>
      )}
    </AnalyticsConsumer>
  );
};

TextNote.propTypes = {
  text: PropTypes.string,
  isHeader: PropTypes.bool,
};

export default TextNote;
