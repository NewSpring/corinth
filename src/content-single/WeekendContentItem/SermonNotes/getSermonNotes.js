import gql from 'graphql-tag';

export default gql`
  query GetSermonNotes($contentID: ID!) {
    node(id: $contentID) {
      id
      ... on WeekendContentItem {
        communicators {
          nickName
          firstName
          lastName
        }
        guestCommunicators
        title
        seriesConnection {
          series {
            title
          }
          itemIndex
          itemCount
        }
        sermonNotes {
          id
          __typename
          ... on TextNote {
            id
            isHeader
            text
          }
          ... on ScriptureNote {
            id
            scripture {
              reference
            }
          }
        }
      }
    }
  }
`;
