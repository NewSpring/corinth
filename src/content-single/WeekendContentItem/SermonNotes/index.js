import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import SermonNotes from './SermonNotes';
import GET_SERMON_NOTES from './getSermonNotes';

const SermonNotesConnected = ({ contentID }) => (
  <Query query={GET_SERMON_NOTES} variables={{ contentID }}>
    {({ data: { node = {} }, loading }) => (
      <SermonNotes isLoading={loading} {...node} />
    )}
  </Query>
);

SermonNotesConnected.propTypes = {
  contentID: PropTypes.string,
};

export default SermonNotesConnected;
