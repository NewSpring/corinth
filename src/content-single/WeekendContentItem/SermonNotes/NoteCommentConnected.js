import React from 'react';
import PropTypes from 'prop-types';
import NoteComment from './NoteComment';

const CustomNoteConnected = ({ parentID, onChange }) => (
  <NoteComment onChange={onChange} />
);

CustomNoteConnected.propTypes = {
  onChange: PropTypes.func,
  parentID: PropTypes.string,
};

export default CustomNoteConnected;
