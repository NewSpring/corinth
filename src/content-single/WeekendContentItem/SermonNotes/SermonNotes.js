import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Share } from 'react-native';
import {
  ActionCard,
  H3,
  H5,
  BodyText,
  FlexedView,
  styled,
  Touchable,
  PaddedView,
  Icon,
} from '@apollosproject/ui-kit';
import { LegalText } from '@apollosproject/ui-scripture';
import { AnalyticsConsumer } from '@apollosproject/ui-analytics';
import TextNote from './TextNote';
import ScriptureNote from './ScriptureNote';
import CustomNote from './CustomNote';

const NoteTypeMapper = (props) => {
  switch (props.type) {
    case 'TextNote':
      return <TextNote {...props} />;
    case 'ScriptureNote':
      return <ScriptureNote {...props} />;
    default:
      return null;
  }
};

NoteTypeMapper.propTypes = {
  type: PropTypes.string,
};

const ExportWrapper = styled({
  flexDirection: 'row',
})(FlexedView);

const SermonNotes = ({ isLoading, ...contentItem }) => {
  const hasNotes = !isLoading && !contentItem.sermonNotes.length;
  const [exports, setExports] = useState([]);
  const handleChangeExportText = (index) => (text) => {
    const changed = exports;
    changed[index] = text;
    setExports(changed);
  };
  const {
    title = '',
    seriesConnection: { series: { title: seriesTitle } } = {
      series: { title },
    },
    communicators = [],
    guestCommunicators = [],
    sermonNotes = [],
  } = contentItem;
  const communicatorNames = communicators.map(
    ({ nickName, firstName, lastName }) =>
      `${nickName || firstName} ${lastName}`
  );
  const speakers = communicatorNames.concat(guestCommunicators);
  const copyrights = new Set(
    sermonNotes
      .filter(({ scripture }) => !!scripture)
      .map((note) => note.scripture.copyright)
  );

  // set up exported text
  useEffect(() => {
    const exportTemplate = [];
    sermonNotes.forEach((note) => {
      exportTemplate.push(note.simpleText);
      exportTemplate.push('');
    });
    setExports(exportTemplate);
  }, []);

  return (
    <ActionCard
      isLoading={isLoading}
      action={
        hasNotes ? (
          <AnalyticsConsumer>
            {({ track }) => (
              <Touchable
                onPress={() => {
                  const textExports = exports.filter((text) => text !== '');
                  Share.share({
                    message: `Sermon Notes\n${title}\n${seriesTitle}${
                      speakers.length ? '\n' : ''
                    }${speakers.join('\n')}${
                      textExports.length ? '\n\n' : ''
                    }${textExports.join('\n\n')}`,
                  });
                  track({
                    eventName: 'Exported Sermon Notes',
                    properties: { title },
                  });
                }}
              >
                <ExportWrapper>
                  <PaddedView vertical={false}>
                    <BodyText>Export</BodyText>
                  </PaddedView>
                  <Icon name={'export'} size={24} />
                </ExportWrapper>
              </Touchable>
            )}
          </AnalyticsConsumer>
        ) : null
      }
    >
      <H3 isLoading={false}>{hasNotes ? 'Sermon Notes' : null}</H3>
      <H5>{title}</H5>
      <H5>{seriesTitle}</H5>
      {speakers.length
        ? speakers
            .filter((speaker) => speaker !== '')
            .map((speaker) => <H5 key={speaker}>{speaker}</H5>)
        : null}

      {sermonNotes.length ? (
        <>
          {sermonNotes.map((note, i) => (
            <>
              <NoteTypeMapper type={note.__typename} key={note.id} {...note} />
              {note.allowsCustomNote ? (
                <CustomNote onChange={handleChangeExportText(i * 2 + 1)} />
              ) : null}
            </>
          ))}
          {[...copyrights].map((text) => (
            <LegalText key={text}>{text}</LegalText>
          ))}
        </>
      ) : null}
    </ActionCard>
  );
};

SermonNotes.propTypes = {
  isLoading: PropTypes.bool,
};

export default SermonNotes;
