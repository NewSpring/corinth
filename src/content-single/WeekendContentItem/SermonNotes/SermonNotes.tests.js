import React from 'react';

import Providers from '../../../Providers';
import { renderWithApolloData } from '../../../utils/testUtils';
import getSermonNotes from './getSermonNotes';
import SermonNotes from './SermonNotes';
import SermonNotesConnected from '.';

const contentItemStuff = {
  id: 'WeekendContentItem:123',
  __typename: 'WeekendContentItem',
  title: 'a sermon',
  seriesConnection: {
    __typename: 'SeriesConnection',
    series: { __typename: 'ContentSeriesContentItem', title: 'a series' },
  },
  communicators: [],
  guestCommunicators: [],
};

const notes = [
  {
    id: 'TextNote:123',
    __typename: 'TextNote',
    simpleText: '1. Point One',
    allowsCustomNote: false,
    isHeader: true,
    hasBlanks: true,
    hiddenText: '1. Point ___',
  },
  {
    id: 'TextNote:234',
    __typename: 'TextNote',
    simpleText: 'Sub point',
    allowsCustomNote: true,
    isHeader: false,
    hasBlanks: false,
    hiddenText: null,
  },
  {
    id: 'ScriptureNote:123',
    __typename: 'ScriptureNote',
    simpleText:
      '11) The greatest among you will be your servant.  Matthew 23:11',
    allowsCustomNote: false,
    scripture: {
      __typename: 'Scripture',
      reference: 'Matthew 23:11',
      html:
        '<p class="p"><span data-number="11" data-sid="MAT 23:11" class="v">11</span><span class="wj">The greatest among you will be your servant.</span> </p>',
      copyright:
        '\n          The Holy Bible, New International Version® NIV®\n          Copyright © 1973, 1978, 1984, 2011 by Biblica, Inc.®\n          Used by Permission of Biblica, Inc.® All rights reserved worldwide.\n        ',
      version: 'NIV',
    },
  },
];

const emptyMock = {
  request: {
    query: getSermonNotes,
    variables: { contentID: 'WeekendContentItem:123' },
  },
  result: {
    data: {
      node: {
        sermonNotes: [],
        ...contentItemStuff,
      },
    },
  },
};

const notesMock = { sermonNotes: notes, ...contentItemStuff };

describe('the sermon notes', () => {
  it('renders a custom note block when notes are blank', async () => {
    const tree = await renderWithApolloData(
      <Providers mocks={[emptyMock]}>
        <SermonNotesConnected contentID={'WeekendContentItem:123'} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('renders sermon notes', async () => {
    const tree = await renderWithApolloData(
      <Providers>
        <SermonNotes isLoading={false} {...notesMock} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
