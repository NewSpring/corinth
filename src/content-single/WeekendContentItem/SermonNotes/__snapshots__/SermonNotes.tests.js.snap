// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the sermon notes renders a custom note block when notes are blank 1`] = `null`;

exports[`the sermon notes renders sermon notes 1`] = `
<View
  isLoading={false}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 16,
      "marginHorizontal": 16,
      "marginVertical": 12,
      "shadowColor": "rgba(8, 0, 20, 0.25)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 8,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 16,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 16,
          "paddingVertical": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#303030",
            "fontFamily": "Colfax-Black",
            "fontSize": 24,
            "lineHeight": 27.6,
          }
        }
      >
        Sermon Notes
      </Text>
      <Text
        style={
          Object {
            "color": "#303030",
            "fontFamily": "Colfax-Medium",
            "fontSize": 14,
            "lineHeight": 20.16,
          }
        }
      >
        a sermon
      </Text>
      <Text
        style={
          Object {
            "color": "#303030",
            "fontFamily": "Colfax-Medium",
            "fontSize": 14,
            "lineHeight": 20.16,
          }
        }
      >
        a series
      </Text>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          horizontal={false}
          style={
            Object {
              "paddingHorizontal": 0,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Bold",
                "fontSize": 16,
                "lineHeight": 23.04,
              }
            }
          >
            1. Point ___
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          horizontal={false}
          style={
            Object {
              "paddingHorizontal": 0,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            bold={false}
            italic={false}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Regular",
                "fontSize": 16,
                "lineHeight": 26,
              }
            }
          >
            Sub point
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          horizontal={false}
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "paddingHorizontal": 0,
              "paddingVertical": 16,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={24}
            bbWidth={24}
            focusable={false}
            height={24}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                undefined,
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={24}
          >
            <RNSVGGroup
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            >
              <RNSVGPath
                d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm6 13h-5v5h-2v-5h-5v-2h5v-5h2v5h5v2z"
                fill={
                  Array [
                    0,
                    4278190080,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={Array []}
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <View
            style={
              Object {
                "paddingHorizontal": 16,
                "paddingVertical": 0,
              }
            }
            vertical={false}
          >
            <Text
              bold={false}
              italic={false}
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "Colfax-Regular",
                  "fontSize": 16,
                  "lineHeight": 26,
                }
              }
            >
              Add a Note
            </Text>
          </View>
        </View>
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Bold",
              "fontSize": 16,
              "lineHeight": 23.04,
              "paddingBottom": 11.52,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Bold",
                "fontSize": 16,
                "lineHeight": 23.04,
              }
            }
          >
            Matthew 23:11
          </Text>
          <Text
            style={
              Object {
                "color": "#505050",
                "fontFamily": "Colfax-Bold",
                "fontSize": 12,
                "lineHeight": 17.28,
              }
            }
          >
             
            NIV
          </Text>
        </Text>
        <View
          style={
            Object {
              "paddingBottom": 17.28,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "paddingBottom": 0,
                }
              }
            >
              <Text
                bold={false}
                italic={false}
                style={
                  Object {
                    "color": "#303030",
                    "fontFamily": "DroidSerif",
                    "fontSize": 16,
                    "lineHeight": 26,
                  }
                }
              >
                <Text
                  bold={false}
                  italic={false}
                  style={
                    Object {
                      "color": "#505050",
                      "fontFamily": "DroidSerif",
                      "fontSize": 9.6,
                      "lineHeight": 26,
                    }
                  }
                >
                   
                  <Text>
                    11
                  </Text>
                  Â 
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#8b0000",
                    }
                  }
                >
                  <Text>
                    The greatest among you will be your servant.
                  </Text>
                </Text>
              </Text>
            </View>
          </View>
        </View>
        <Text
          bold={false}
          italic={false}
          style={
            Object {
              "color": "#505050",
              "fontFamily": "Colfax-Regular",
              "fontSize": 9.6,
              "lineHeight": 10.37,
            }
          }
        />
      </View>
      <Text
        bold={false}
        italic={false}
        style={
          Object {
            "color": "#505050",
            "fontFamily": "Colfax-Regular",
            "fontSize": 9.6,
            "lineHeight": 10.37,
          }
        }
      >
        
          The Holy Bible, New International VersionÂ® NIVÂ®
          Copyright Â© 1973, 1978, 1984, 2011 by Biblica, Inc.Â®
          Used by Permission of Biblica, Inc.Â® All rights reserved worldwide.
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-end",
          "paddingHorizontal": 16,
          "paddingTop": 0,
          "paddingVertical": 16,
        }
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 16,
                "paddingVertical": 0,
              }
            }
            vertical={false}
          >
            <Text
              bold={false}
              italic={false}
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "Colfax-Regular",
                  "fontSize": 16,
                  "lineHeight": 26,
                }
              }
            >
              Export
            </Text>
          </View>
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={24}
            bbWidth={24}
            focusable={false}
            height={24}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                undefined,
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={512}
            vbWidth={576}
            width={24}
          >
            <RNSVGGroup
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            >
              <RNSVGPath
                d="M572.29 279.06l-71.77-72.31c-7.55-7.6-20.52-2.26-20.52 8.45v48.88h-96v-132.1c0-12.7-5.17-25-14.17-33.99L285.94 14.1c-9-9-21.2-14.1-33.89-14.1H47.99C21.5.1 0 21.6 0 48.09v415.92C0 490.5 21.5 512 47.99 512h287.94c26.5 0 48.07-21.5 48.07-47.99V360.07h-48v103.94H47.99V48.09h159.97v103.98c0 13.3 10.7 23.99 24 23.99H336v88.01H172c-6.63 0-12 5.37-12 12v24c0 6.63 5.37 12 12 12h308v48.88c0 10.71 12.97 16.05 20.52 8.45l71.77-72.31c4.95-4.99 4.95-13.04 0-18.03zM255.95 128.07V51.99l76.09 76.08h-76.09z"
                fill={
                  Array [
                    0,
                    4281348144,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</View>
`;
